woocommerce/woocommerce-gutenberg-products-block
# Add global styles to Product Title block

Fixes #4919

This PR:
- Replaces custom text color and font size with [Block Supports](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/).
- Adds global style support.
- Update the title block to use [`apiVersion: 2`](https://make.wordpress.org/core/2020/11/18/block-api-version-2/).




### Screenshots

Block setting:

![image](https://user-images.githubusercontent.com/5423135/139051597-fa60a510-80db-4288-9b8d-e99a63ed8913.png)

Global style:

![image](https://user-images.githubusercontent.com/5423135/139051765-263de5cc-5a23-419c-a13b-6f932b2a1111.png)




### Testing

#### Automated Tests
### Manual Testing

How to test the changes in this Pull Request:

0. Activate and update Gutenberg to the latest `trunk` commit.
1. In a classic theme (ie: Storefront):
	1. Add the `All Products` block to a post or page.
	2. Verify you can change the text color and font size of the loop item title.
	3. Verify that the style work on both front-end and the block's View mode/Edit mode.
2. In a block-based theme (ie: [TT1 Blocks](https://wordpress.org/themes/tt1-blocks/)):
	1. Go to Appearance > Site editor, click on the Styles icon and verify the Product Title block is shown and you can tweak its styles:
![image](https://user-images.githubusercontent.com/5423135/139050781-9647d491-0a20-46cd-9862-779c8822522a.png)
	2. Add the All Products block to a post or page.
	3. Verify it honors the styles you set in the Site editor.
	4. Change the styles in the post/page editor and verify they have priority over the styles from the Site editor.



### Changelog

> Added global styles (text color and font size) to Product Title block.
